auth/data/src/main/java/com/pronoidsoftware/auth/data/di/AuthRepositoryModule.kt

```diff
@@ -0,0 +1,20 @@
+package com.pronoidsoftware.auth.data.di
+
+import com.pronoidsoftware.auth.data.AuthRepositoryImplementation
+import com.pronoidsoftware.auth.domain.AuthRepository
+import dagger.Binds
+import dagger.Module
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+abstract class AuthRepositoryModule {
+
+    @Binds
+    @Singleton
+    abstract fun bindAuthRepository(
+        authRepositoryImplementation: AuthRepositoryImplementation,
+    ): AuthRepository
+}
```

Comment:
Calling this `AuthRepositoryModule` kind of communicates that it may not contain anything else other than the `AuthRepository`. Could it make sense to give it a more generic name, so it can also contain other auth related abstractions? (I've also always found it difficult to give abstract modules a different/better name than normal modules with `@Provides` 😅). 
But from my side something like `AbstractAuthModule` is great and communicates it all.